### PR TITLE
SONARJAVA-4863 S2092: Support detection of missing secure cookie flag for Spring

### PR DESCRIPTION
**Pull request by irina-batinic-sonarsource**

### CODE REVIEW

File: java-checks/src/test/java/org/sonar/java/checks/security/SecureCookieCheckTest.java [hunk 1]
```diff
@@ -19,17 +19,22 @@
...
 import static org.sonar.java.checks.verifier.TestUtils.mainCodeSourcesPath;
+import static org.sonar.java.checks.verifier.TestUtils.nonCompilingTestSourcesPath;
 
 class SecureCookieCheckTest {
 
   private static final String SOURCE_PATH = "checks/security/SecureCookieCheck.java";
   private static final String TEST_SOURCE_PATH = mainCodeSourcesPath(SOURCE_PATH);
   private static final String NON_COMPILING_TEST_SOURCE_PATH = nonCompilingTestSourcesPath(SOURCE_PATH);
+  private static final String SPRING_3_2_4 = "../java-checks-test-sources/spring-3.2.4";
+  public static final List<File> SPRING_3_2_4_CLASSPATH = TestClasspathUtils.loadFromFile(SPRING_3_2_4 + "/target/test-classpath.txt");
```
[the remark this answers is not in the document]
Comment (irina-batinic-sonarsource) [reply java-checks/src/test/java/org/sonar/java/checks/security/SecureCookieCheckTest.java]: Good idea!
It is done.

File: java-checks-testkit/src/main/java/org/sonar/java/checks/Constants.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.java.checks;
+
+public class Constants {
+
+  private Constants() {
+  }
+
+  public static final String SPRING_3_2_4 = "../java-checks-test-sources/spring-3.2.4";
```
[the remark this answers is not in the document]
Comment (irina-batinic-sonarsource) [reply java-checks-testkit/src/main/java/org/sonar/java/checks/Constants.java:27]: I think it is important to leave a major.minor version. We don't need a patch number.
During the investigation, I noticed new classes and methods were added and removed in minor versions. So we might need different versions.

File: java-checks/src/test/java/org/sonar/java/checks/security/SecureCookieCheckTest.java [hunk 3]
```diff
@@ -19,17 +19,23 @@
...
 import static org.sonar.java.checks.verifier.TestUtils.mainCodeSourcesPath;
+import static org.sonar.java.checks.verifier.TestUtils.mainCodeSourcesPathInModule;
+import static org.sonar.java.checks.verifier.TestUtils.nonCompilingTestSourcesPath;
 
 class SecureCookieCheckTest {
 
   private static final String SOURCE_PATH = "checks/security/SecureCookieCheck.java";
   private static final String TEST_SOURCE_PATH = mainCodeSourcesPath(SOURCE_PATH);
   private static final String NON_COMPILING_TEST_SOURCE_PATH = nonCompilingTestSourcesPath(SOURCE_PATH);
+  public static final List<File> SPRING_3_2_4_CLASSPATH = TestClasspathUtils.loadFromFile(Constants.SPRING_3_2_CLASSPATH);
```
Comment (johann-beleites-sonarsource) [inline java-checks/src/test/java/org/sonar/java/checks/security/SecureCookieCheckTest.java:38]:
```suggestion
  public static final List<File> SPRING_3_2_CLASSPATH = TestClasspathUtils.loadFromFile(Constants.SPRING_3_2_CLASSPATH);
```

File: java-checks/src/test/java/org/sonar/java/checks/security/SecureCookieCheckTest.java [hunk 4]
```diff
@@ -54,4 +60,14 @@ void test_non_compiling() {
       .withCheck(new SecureCookieCheck())
       .verifyIssues();
   }
+
+  @Test
+  void test_with_spring_3_2_4() {
+    CheckVerifier.newVerifier()
+      .onFile(mainCodeSourcesPathInModule(Constants.SPRING_3_2, "checks/SecureCookieCheckSample.java"))
+      .withCheck(new SecureCookieCheck())
+      .withClassPath(SPRING_3_2_4_CLASSPATH)
```
Comment (johann-beleites-sonarsource) [inline java-checks/src/test/java/org/sonar/java/checks/security/SecureCookieCheckTest.java:69]:
```suggestion
      .withClassPath(SPRING_3_2_CLASSPATH)
```

File: java-checks/src/test/java/org/sonar/java/checks/security/SecureCookieCheckTest.java [hunk 5]
```diff
@@ -54,4 +60,14 @@ void test_non_compiling() {
       .withCheck(new SecureCookieCheck())
       .verifyIssues();
   }
+
+  @Test
+  void test_with_spring_3_2_4() {
```
Comment (johann-beleites-sonarsource) [inline java-checks/src/test/java/org/sonar/java/checks/security/SecureCookieCheckTest.java:65]:
```suggestion
  void test_with_spring_3_2() {
```

**Review by johann-beleites-sonarsource — APPROVED** [review]
Nice!

**Pull request merged**